packages/contracts-bedrock/test/invariants/symbolic/MockL2ToL2Messenger.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import "src/L2/L2ToL2CrossDomainMessenger.sol";
+import "forge-std/Test.sol";
+
+contract MockL2ToL2Messenger is IL2ToL2CrossDomainMessenger {
+    uint256 internal messageNonce;
+    address internal immutable CURRENT_XDOMAIN_SENDER;
+
+    constructor(address _currentXDomSender) {
+        // Setting the current cross domain sender for the check of sender address equals the supertoken address
+        CURRENT_XDOMAIN_SENDER = _currentXDomSender;
```

Comment:
shouldn't this be the sender of a given sendMessage instead (same in Medusa)?

Reply:
since this is stateless, I only need this to be the supertoken so it doesn't revert.
But I see more value on adding it for Medusa.
This also makes me think that a property regarding the cross domain sender is missing

Reply:
Fuck, there is on (id 7) and I confused it a bit. Will hold that test but also adding the `xDomainSender` caller invariant